test/integration/targets/ini_file/tasks/main.yml

```diff
@@ -0,0 +1,153 @@
+---
+# test code for ini_file plugins
+# (c) 2017 Red Hat Inc.
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+- name: record the output directory
+  set_fact: output_file={{output_dir}}/foo.ini
+
+- name: add "fav=lemonade" is in section "[drinks]" in specified file
+  ini_file:
+    path: "{{ output_file }}"
+    section: drinks
+    option: fav
+    value: lemonade
+  register: result1
+
+- name: verify ini_file 'changed' is true
+  assert:
+    that:
+      - result1.changed == True
+      - result1.msg == 'section and option added'
+
+# sha1sum: df66db77ecc39daa88abd1aac930dd6b762547b4'
+# expected content:
+#"""
+#
+#[drinks]
+#fav = lemonade
+#"""
+
+- name: stat the file to compute checksum
+  stat:
+    path: "{{ output_file }}"
+  register: resultstat1
+
+- name: Verify content of ini file is as expected (checksum)
+  assert:
+    that:
+      - resultstat1.stat.checksum == 'df66db77ecc39daa88abd1aac930dd6b762547b4'
```

Comment:
Instead of putting the expected content in comments and comparing by checksum, perhaps we can set the expected content in a fact and compare the actual contents instead. That should be more maintainable (no checksums to compute) and make failures much more obvious (no more "hey, these checksums don't match, why?")

I think [literal style multi-line strings](http://stackoverflow.com/a/21699210) would work well for this.

Yes, we have other tests that use checksums for comparisons like this, but they're not user friendly.